test/common/netlib.py

```diff
@@ -205,12 +213,25 @@ def slow_down_dhclient(self, delay):
         """)
         self.addCleanup(self.machine.execute, "umount /usr/sbin/dhclient")
 
-    def wait_onoff(self, sel, val):
+    def wait_onoff(self, sel: str, *, val: bool) -> None:
         self.browser.wait_visible(sel + " input[type=checkbox]" + (":checked" if val else ":not(:checked)"))
 
-    def toggle_onoff(self, sel):
+    def toggle_onoff(self, sel: str) -> None:
         self.browser.click(sel + " input[type=checkbox]")
 
-    def login_and_go(self, *args, **kwargs):
-        super().login_and_go(*args, **kwargs)
+    def login_and_go(
+        self,
+        path: str | None = None,
+        *,
+        user: str | None = None,
+        password: str | None = None,
+        host: str | None = None,
+        superuser: bool = True,
+        urlroot: str | None = None,
+        tls: bool = False,
+        enable_root_login: bool = False
+    ) -> None:
+        super().login_and_go(path=path, user=user, password=password,
+                             host=host, superuser=superuser, urlroot=urlroot,
+                             tls=tls, enable_root_login=enable_root_login)
         self.nm_checkpoints_disable()
```

Comment:
Ugh, we do that? That feels highly non-obvious. But indeed not the concern of this PR.

Reply:
My point is that I don't like the duplicate definition of `browser: Browser` and `machine: Machine`.  If we want to use those things then we should simply derive from the class that already has those things, instead of deriving from a different class and then adding those things in for ourselves.

Reply:
I agree this is non-obvious and not nice, but also didn't want to fix it in this PR